freva-client/src/freva_client/query.py

```diff
@@ -757,3 +830,51 @@ def _get(
                 raise ValueError(msg) from None
             logger.warning(msg)
         return None
+
+    def _put(
+        self, url: str, data: Dict[str, Any], **kwargs: Any
+    ) -> Optional[requests.models.Response]:
+        """Apply the PUT method to the databrowser."""
+        logger.debug(
+            "PUT request to %s with data: %s and parameters: %s",
+            url,
+            data,
+            self._params,
+        )
+        kwargs.setdefault("timeout", 30)
+        params = kwargs.pop("params", {})
+        try:
+            res = requests.put(
+                url, json=data, params={**self._params, **params}, **kwargs
+            )
+            res.raise_for_status()
+            return res
+        except KeyboardInterrupt:
+            pprint("[red][b]User interrupt: Exit[/red][/b]", file=sys.stderr)
+
+        except (
+            requests.exceptions.ConnectionError,
+            requests.exceptions.HTTPError,
+        ) as error:
+            msg = f"adding user data request failed with {error}"
+            logger.warning(msg)
```

Comment:
I am not sure, how we want to deal with errors. In some instances, it might useful to throw an error, in some not - think about how you want to deal with that.